pkg/pod/entrypoint.go

```diff
@@ -333,9 +377,13 @@ func TrimSidecarPrefix(name string) string { return strings.TrimPrefix(name, sid
 
 // StepName returns the step name after adding "step-" prefix to the actual step name or
 // returns "step-unnamed-<step-index>" if not specified
-func StepName(name string, i int) string {
+func StepName(name string, i int, addPrefix bool) string {
+	prefix := ""
+	if addPrefix {
+		prefix = stepPrefix
+	}
 	if name != "" {
-		return fmt.Sprintf("%s%s", stepPrefix, name)
+		return fmt.Sprintf("%s%s", prefix, name)
 	}
-	return fmt.Sprintf("%sunnamed-%d", stepPrefix, i)
+	return fmt.Sprintf("%sunnamed-%d", prefix, i)
 }
```

Comment:
I don't like overloading this function like this - it makes it not clear how callers are supposed to use it, and this is important because this is what is generating the step container name.

I'd recommend breaking the new behavior into a new function - e.g.

```go
func StepName(name string, i int, addPrefix bool) string {
	return fmt.Sprintf("%s%s", prefix, normalizeStepName(name, i))
}

func normalizeStepName(name string, i int) string  {
	if name != "" {
		return name
	}
	return fmt.Sprintf("unnamed-%d", stepPrefix, i)
}
```